### ISSUE TITLE
Primary Set-Up UI: "Specifications" Section

### ISSUE THREAD

**Issue by acshields** [topcoder-platform/work-manager#737]
### Section Display Order
- "Specifications" section is displayed after the "Participants"
- "Specifications" section is displayed before the "Timeline"

### **Header of Section**
- Section Title: "Specifications"

### **Main Body Question/Answers**
_Field 1 is a required field._
_Field 1:_ Technology Tags
_Field 1 Answer:_ [Responsive Text Entry]

_Field 2 is a required field._
_Field 2:_ Public Specification
_Field 2 Sub-Text 1:_ This specification will be viewable on public listings.
_Field 2 Sub-Text 2 Hyperlink_: Reference specification templates
_Field 2 Answer:_ [Markdown-Enabled Text Field]

_Field 3 is an optional field._
If the user clicks the "+ Add Private Specification" button, which displays beneath the public specification, then the user will see Field 3 presented as follows:
_Field 3:_ Private Specification
_Field 3 Sub-Text:_ This specification will only be viewable to registrants.
_Field 3 Sub-Text 2 Hyperlink_: Reference specification templates
_Field 3 Answer:_ [Markdown-Enabled Text Field]

_Field 4 is an optional field._
_Field 4:_ File Attachments
_Field 4 Answer:_ Filestack upload workflow.

**Comment by maxceem** [comment]
1. Issue mention **Deliverable Format** field, but the design doesn't have it. 
    1. Should we show it or no?
    2. If we should show it. At the moment we are showing delivery types using the UI shown below. Should we replace it with a simple text field?

        ![image](https://user-images.githubusercontent.com/146016/99960848-17a58580-2d96-11eb-9196-8ce27146cb93.png)

2. Which field in the data in Challenge API should we use to store value for **Branding Guidlines Required**?

     ![image](https://user-images.githubusercontent.com/146016/99961263-c1851200-2d96-11eb-96cb-51d04a63ee09.png)

    Here is how we store other values:

    ![image](https://user-images.githubusercontent.com/146016/99961334-e8dbdf00-2d96-11eb-8e49-092c707a4da1.png)

3. Which API can we use to handle file uploading? 

    ![image](https://user-images.githubusercontent.com/146016/99961484-35bfb580-2d97-11eb-9e0a-9fe1d3d3611c.png)

**Comment by vikasrohit** [comment]
@maxceem 

1. I think we can keep it somewhere in the form with tag styles matching with the designs. Something like:

<img width="780" alt="Screen Shot 2020-11-23 at 6 07 34 PM" src="https://user-images.githubusercontent.com/2417632/99962760-c5289100-2db6-11eb-8d39-79ee21b6d0ef.png">

2. @rootelement need your help

3. We need to use Filestack for uploading the files and challenge api should be providing the feature of attachments already. @rootelement do you think we are ready for attachments now?

**Comment by acshields** [comment]
@vikasrohit @maxceem 

For #1, I am good with your suggestion Vikas.

**Comment by vikasrohit** [comment]
Thanks @acshields I have kept the issue in `Requirement Clarification` stage because we are still waiting on @rootelement for status of attachment feature in the api. If it is stable and we are planning to release it to prod soon, we can start working on the UI part as well. We have created separate issue https://github.com/topcoder-platform/work-manager/issues/917 for tracking the feature of supporting attachments without new designs.

**Comment by ThomasKranitsas** [comment]
@vikasrohit as you correctly said, we need to use Filestack for uploading the files. We have updated the challenge API to support CRUD operation for attachments (`CREATE` only links an already uploaded file with a challenge, it doesn't actually upload the file).

It's been deployed on dev and I'm actively doing testing there. Feel free to also test it out. The latest swagger has all the info

**Comment by acshields** [comment]
@vikasrohit - Adding this to the December project in "Requirements Clarification" then, since I am assuming it won't be done/released by tomorrow (for tracking purposes).

**Comment by maxceem** [comment]
@ThomasKranitsas @vikasrohit could you please, let me know what are whitelisted S3 buckets for the challenge API?

When I'm trying to create an attachment with the body:
```js
{
   fileSize: 4
   name: "test2.txt" 
   url: "https://topcoder-dev-media.s3.amazonaws.com/challenges/2608c97a-5704-40a3-bff2-bbd77c872061/SPECIFICATION_ATTACHMENTS/Hd35NbqpSjqPQga8H2WW_test2.txt"
}
```
I'm getting an error:
```
The bucket topcoder-dev-media is not in the whitelist
```

**Comment by maxceem** [comment]
@ThomasKranitsas I've also tried with another bucket shared by Vikas with me `tc-challenge-v5-dev` but I'm still getting `The bucket tc-challenge-v5-dev is not in the whitelist`.

Could you advise which bucket we can use on DEV? 

Here is the request I'm sending, just in case:
```
curl 'https://api.topcoder-dev.com/v5/challenges/2608c97a-5704-40a3-bff2-bbd77c872061/attachments' \
  -H 'authority: api.topcoder-dev.com' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'authorization: Bearer <ADMIN_TOKEN> \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.67 Safari/537.36' \
  -H 'content-type: application/json' \
  -H 'origin: http://localhost:3000' \
  -H 'sec-fetch-site: cross-site' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-dest: empty' \
  -H 'referer: http://localhost:3000/' \
  -H 'accept-language: ru-UA,ru;q=0.9,en-US;q=0.8,en;q=0.7' \
  --data-binary '{"name":"test1.txt","fileSize":4,"url":"https://tc-challenge-v5-dev.s3.amazonaws.com/challenges/2608c97a-5704-40a3-bff2-bbd77c872061/SPECIFICATION_ATTACHMENTS/gPdtJTimRMy9MXyVI343_test1.txt"}' \
  --compressed
```

**Comment by vikasrohit** [comment]
@maxceem did we got this problem solved?

**Comment by maxceem** [comment]
@vikasrohit yes, Thomas and James helped me.

**Comment by vikasrohit** [comment]
@ajefts closing this and similar for now to clean up things.
